Test: Adding integration tests for cv_configlet_v3

Shivani-chourasiya opened the pull request:
## Change Summary

Adding integration tests for cv_configlet_v3

## Related Issue(s)

Fixes #<ISSUE ID>

## Component(s) name

`arista.cvp.cv_configlet_v3`

## How to test

run `molecule converge -s cv_configlet_v3`

## Checklist

### User Checklist

<!-- Add your own checklist using MD syntax and by replacing N/A -->
- N/A

### Repository Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly. (check the box if not applicable)
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

Shivani-gslab committed:
Test: Adding integration tests for cv_configlet_v3

Shivani-gslab committed:
Added tests for configlet updation and deleting non-existing configlet

Shivani-gslab committed:
Added tests for configlet updation and deleting non-existing configlet

sugetha24 commented on ansible_collections/arista/cvp/molecule/cv_configlet_v3/test_cv_configlet_v3.yaml:
```diff
@@ -0,0 +1,93 @@
+          - CV_CONFIGLET_V3_RESULT.configlets_created.configlets_created_list == ['configlet1', 'configlet2']
+          - CV_CONFIGLET_V3_RESULT.configlets_created.success == true
+
+    - name: "Push already existing config"
```
This will change the config `configlet1` and `configlet2` since we are using `random` in line8 and line9. This is a good test, but it's testing changed configuration. Update the name accordingly.
Also, We need to have another test where wee are pushing configlet with same configuration.

Shivani-chourasiya replied:
fixed!

Shivani-gslab committed:
Addressing review comments

sugetha24 commented on ansible_collections/arista/cvp/molecule/cv_configlet_v3/test_cv_configlet_v3.yaml:
```diff
@@ -0,0 +1,109 @@
+  vars:
+    CVP_CONFIGLETS:
+      configlet1: 'alias a{{ 999 | random }} show lldp'
+      configlet2: 'This is second configlet'
```
```suggestion
      configlet2: '! This is second configlet'
```

noredistribution committed:
Update test_cv_configlet_v3.yaml

Update regex to only match on configlet1 and configlet2, otherwise if you have a system with other configlets that have the string "configlet" the assert will fail

noredistribution reviewed (approved):
LGTM!

sugetha24 commented on ansible_collections/arista/cvp/molecule/cv_configlet_v3/test_cv_configlet_v3.yaml:
```diff
@@ -0,0 +1,109 @@
+      arista.cvp.cv_facts_v3:
+        facts:
+          - configlets
+        regexp_filter: "configlet"
```
@noredistribution shouldn't we update the regexp here too?

noredistribution committed:
Update test_cv_configlet_v3.yaml

update regex to only match on configlet1 and 2

sugetha24 committed:
Update ansible_collections/arista/cvp/molecule/cv_configlet_v3/test_cv_configlet_v3.yaml

noredistribution reviewed (approved):
re-tested and works well! LGTM! Thank you!
